src/numpy-stubs/_typing/_ufunc.pyi

```diff
@@ -222,6 +222,72 @@ class _Call11Bool(Protocol):
     ) -> NDArray[np.bool] | np.bool: ...
 
 @type_check_only
+class _Call11Logical(Protocol):
+    @overload
+    def __call__(  # (scalar, dtype: np.object_) -> bool
+        self,
+        x: _ScalarLike_co,
+        /,
+        dtype: _DTypeLike[np.object_],
+        out: None = None,
+        **kwargs: Unpack[_Kwargs2],
+    ) -> bool: ...
+    @overload
+    def __call__(  # (scalar) -> np.bool
+        self,
+        x: _NumberLike_co,
+        /,
+        out: None = None,
+        dtype: _DTypeLikeBool | None = None,
+        **kwargs: Unpack[_Kwargs2],
+    ) -> np.bool: ...
+    @overload
+    def __call__(  # (array-like, dtype: np.object_) -> np.object_
+        self,
+        x: _ArrayLikeNumber_co | _ArrayLikeObject_co,
+        /,
+        dtype: _DTypeLike[np.object_],
+        out: None = None,
+        **kwargs: Unpack[_Kwargs2],
+    ) -> NDArray[np.object_] | bool: ...
+    @overload
+    def __call__(  # (array-like, out: T) -> T
+        self,
+        x: _ArrayLikeNumber_co,
```

Comment:
Here, `out` is provided, and it could be a `NDArray[object_]`. So it should probably also accept object array-likes:

```suggestion
        x: _ArrayLikeNumber_co | _ArrayLikeObject_co,
```